src/ngx_http_lua_api.c

```diff
@@ -194,13 +198,12 @@ ngx_http_lua_shared_memory_init(ngx_shm_zone_t *shm_zone, void *data)
 
     lmcf->shm_zones_inited++;
 
-    if (lmcf->shm_zones_inited == lmcf->shm_zones->nelts
-        && lmcf->init_handler)
+    if (lmcf->shm_zones_inited == lmcf->shm_zones->nelts)
     {
         saved_cycle = ngx_cycle;
         ngx_cycle = ctx->cycle;
 
-        rc = lmcf->init_handler(ctx->log, lmcf, lmcf->lua);
+        rc = ngx_http_lua_delay_init_handler();
```

Comment:
@rainingmaster That does not sound right to me. The 3rd-party modules can do whatever it likes, which may have nothing to do with ngx_lua's own `lua_shared_dict` directive. Or your change here is only meant for `ngx_meta_lua_module` and `lua_shared_dict`? That is, not meant for general 3rd-party modules?